src/gui/vector/qgsattributesformmodel.h

```diff
@@ -0,0 +1,459 @@
+
+        /**
+         * Returns the QML editor configuration.
+         *
+         * \see setQmlElementEditorConfiguration()
+         */
+        QmlElementEditorConfiguration qmlElementEditorConfiguration() const;
+
+        /**
+         * Sets the QML editor configuration.
+         *
+         * \see qmlElementEditorConfiguration()
+         */
+        void setQmlElementEditorConfiguration( const QmlElementEditorConfiguration &qmlElementEditorConfiguration );
+
+        /**
+         * Returns the HTML editor configuration.
+         *
+         * \see setHtmlElementEditorConfiguration()
+         */
+        HtmlElementEditorConfiguration htmlElementEditorConfiguration() const;
+
+        /**
+         * Sets the HTML editor configuration.
+         *
+         * \see htmlElementEditorConfiguration()
+         */
+        void setHtmlElementEditorConfiguration( const HtmlElementEditorConfiguration &htmlElementEditorConfiguration );
+
+        /**
+         * Returns the spacer element configuration
+         * \since QGIS 3.30
+         */
+        SpacerElementEditorConfiguration spacerElementEditorConfiguration() const;
+
+        /**
+         * Sets the the spacer element configuration to \a spacerElementEditorConfiguration
+         * \since QGIS 3.30
+         */
+        void setSpacerElementEditorConfiguration( SpacerElementEditorConfiguration spacerElementEditorConfiguration );
+
+        QColor backgroundColor() const;
+        void setBackgroundColor( const QColor &backgroundColor );
+
+        /**
+         * Returns the editor configuration for text element.
+         * \since QGIS 3.30
+         */
+        TextElementEditorConfiguration textElementEditorConfiguration() const;
+
+        /**
+         * Sets the editor configuration for text element to \a textElementEditorConfiguration.
+         * \since QGIS 3.30
+         */
+        void setTextElementEditorConfiguration( const TextElementEditorConfiguration &textElementEditorConfiguration );
+
+      private:
+        int mColumnCount = 1;
+        Qgis::AttributeEditorContainerType mContainerType = Qgis::AttributeEditorContainerType::Tab;
+        bool mShowLabel = true;
+        int mHorizontalStretch = 0;
+        int mVerticalStretch = 0;
+        QgsOptionalExpression mVisibilityExpression;
+        RelationEditorConfiguration mRelationEditorConfiguration;
+        QmlElementEditorConfiguration mQmlElementEditorConfiguration;
+        HtmlElementEditorConfiguration mHtmlElementEditorConfiguration;
+        TextElementEditorConfiguration mTextElementEditorConfiguration;
+        SpacerElementEditorConfiguration mSpacerElementEditorConfiguration;
+        QColor mBackgroundColor;
+        bool mCollapsed = false;
+        QgsOptionalExpression mCollapsedExpression;
+        QgsAttributeEditorElement::LabelStyle mLabelStyle;
+    };
+};
+
+
+class AttributeFormTreeItem
```

Comment:
We'll be relying on this `AttributeFormTreeItem` to hold the underlying parent-child relationships between items/nodes.